diff --git a/dist/index.mjs b/dist/index.mjs
index 110f37ec18c98b1d55ae2b73cc716194e6f9094d..3ea0fadd783f334db71266e45babdcce11076974 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -448,7 +448,7 @@ function convertToGoogleGenerativeAIMessages(prompt, options) {
 
 // src/get-model-path.ts
 function getModelPath(modelId) {
-  return modelId.includes("/") ? modelId : `models/${modelId}`;
+  return `models/${modelId}`;
 }
 
 // src/google-generative-ai-options.ts
