diff --git a/sdk.mjs b/sdk.mjs
index e2dbafb4e2faa1bf2b6b02f0009a2b9bbf57c757..3f07a1d5c2949a246fe5414e69ab45942fa605a2 100644
--- a/sdk.mjs
+++ b/sdk.mjs
@@ -6355,11 +6355,11 @@ class ProcessTransport {
         prompt,
         additionalDirectories = [],
         cwd,
-        executable = isRunningWithBun() ? "bun" : "node",
+        executable = process.execPath,
         executableArgs = [],
         extraArgs = {},
         pathToClaudeCodeExecutable,
-        env = { ...process.env },
+        env = { ...process.env, ELECTRON_RUN_AS_NODE: '1' },
         stderr,
         customSystemPrompt,
         appendSystemPrompt,
