diff --git a/src/markdown-loader.js b/src/markdown-loader.js
index eaf30b114a273e68abbb92c8b07018495e63f4cb..4b06519bdb51845e4693fe877da9de01c7a81039 100644
--- a/src/markdown-loader.js
+++ b/src/markdown-loader.js
@@ -21,7 +21,7 @@ export class MarkdownLoader extends BaseLoader {
             ? (await getSafe(this.filePathOrUrl, { format: 'buffer' })).body
             : await streamToBuffer(fs.createReadStream(this.filePathOrUrl));
         this.debug('MarkdownLoader stream created');
-        const result = micromark(buffer, { extensions: [gfm(), mdxJsx()], htmlExtensions: [gfmHtml()] });
+        const result = micromark(buffer, { extensions: [gfm()], htmlExtensions: [gfmHtml()] });
         this.debug('Markdown parsed...');
         const webLoader = new WebLoader({
             urlOrContent: result,
