diff --git a/es/dropdown/dropdown.js b/es/dropdown/dropdown.js
index 2e45574398ff68450022a0078e213cc81fe7454e..58ba7789939b7805a89f92b93d222f8fb1168bdf 100644
--- a/es/dropdown/dropdown.js
+++ b/es/dropdown/dropdown.js
@@ -2,7 +2,7 @@
 
 import * as React from 'react';
 import LeftOutlined from "@ant-design/icons/es/icons/LeftOutlined";
-import RightOutlined from "@ant-design/icons/es/icons/RightOutlined";
+import { ChevronRight } from 'lucide-react';
 import classNames from 'classnames';
 import RcDropdown from 'rc-dropdown';
 import useEvent from "rc-util/es/hooks/useEvent";
@@ -160,8 +160,10 @@ const Dropdown = props => {
         className: `${prefixCls}-menu-submenu-arrow`
       }, direction === 'rtl' ? (/*#__PURE__*/React.createElement(LeftOutlined, {
         className: `${prefixCls}-menu-submenu-arrow-icon`
-      })) : (/*#__PURE__*/React.createElement(RightOutlined, {
-        className: `${prefixCls}-menu-submenu-arrow-icon`
+      })) : (/*#__PURE__*/React.createElement(ChevronRight, {
+        size: 16,
+        strokeWidth: 1.8,
+        className: `${prefixCls}-menu-submenu-arrow-icon lucide-custom`
       }))),
       mode: "vertical",
       selectable: false,
diff --git a/es/select/useIcons.js b/es/select/useIcons.js
index 572aaaa0899f429cbf8a7181f2eeada545f76dcb..4e175c8d7713dd6422f8bcdc74ee671a835de6ce 100644
--- a/es/select/useIcons.js
+++ b/es/select/useIcons.js
@@ -4,10 +4,10 @@ import * as React from 'react';
 import CheckOutlined from "@ant-design/icons/es/icons/CheckOutlined";
 import CloseCircleFilled from "@ant-design/icons/es/icons/CloseCircleFilled";
 import CloseOutlined from "@ant-design/icons/es/icons/CloseOutlined";
-import DownOutlined from "@ant-design/icons/es/icons/DownOutlined";
 import LoadingOutlined from "@ant-design/icons/es/icons/LoadingOutlined";
 import SearchOutlined from "@ant-design/icons/es/icons/SearchOutlined";
 import { devUseWarning } from '../_util/warning';
+import { ChevronDown } from 'lucide-react';
 export default function useIcons({
   suffixIcon,
   clearIcon,
@@ -54,8 +54,10 @@ export default function useIcons({
           className: iconCls
         }));
       }
-      return getSuffixIconNode(/*#__PURE__*/React.createElement(DownOutlined, {
-        className: iconCls
+      return getSuffixIconNode(/*#__PURE__*/React.createElement(ChevronDown, {
+        size: 16,
+        strokeWidth: 1.8,
+        className: `${iconCls} lucide-custom`
       }));
     };
   }
