diff --git a/core.js b/core.js
index 30c91e66bf595a66c09eb3dbcbda7d58154865f5..b511ff24ea1891904c60174c6ed26ecdd4d5ac51 100644
--- a/core.js
+++ b/core.js
@@ -156,7 +156,7 @@ class APIClient {
             Accept: 'application/json',
             'Content-Type': 'application/json',
             'User-Agent': this.getUserAgent(),
-            ...getPlatformHeaders(),
+            // ...getPlatformHeaders(),
             ...this.authHeaders(opts),
         };
     }
diff --git a/core.mjs b/core.mjs
index ac267bcfcff44b1f7c9bea5513bba94726a31795..dd5bd9f29609d3f0eea4bd5b225f302893df14ad 100644
--- a/core.mjs
+++ b/core.mjs
@@ -149,7 +149,7 @@ export class APIClient {
             Accept: 'application/json',
             'Content-Type': 'application/json',
             'User-Agent': this.getUserAgent(),
-            ...getPlatformHeaders(),
+            // ...getPlatformHeaders(),
             ...this.authHeaders(opts),
         };
     }
